homeassistant/components/media_player/androidtv.py

```diff
@@ -0,0 +1,413 @@
+
+            elif self._available:
+                _LOGGER.error(
+                    "ADB server not connected to {}".format(self._name))
+                _LOGGER.warning(
+                    "Device {} became unavailable.".format(self._name))
+                self._available = False
+
+    def get_state(self, powerOutput, audioOutput):
+        """Process sys output and return the device state."""
+        if 'Display Power: state=ON' not in powerOutput:
+            return STATE_OFF
+        elif 'started' in audioOutput:
+            state = STATE_PLAYING
+        elif 'paused' in audioOutput:
+            state = STATE_PAUSED
+        else:
+            state = STATE_IDLE
+
+        return state
+
+    def get_audio(self, audioOutput):
+        """Process sys output and return the volume, muted state and device."""
+        import re
+        blockPattern = 'STREAM_MUSIC(.*?)- STREAM'
+        streamBlock = re.findall(
+            blockPattern, audioOutput, re.DOTALL | re.MULTILINE)[0]
+
+        devicePattern = 'Devices: (.*?)\W'
+        device = re.findall(devicePattern, streamBlock,
+                            re.DOTALL | re.MULTILINE)[0]
+
+        mutedPattern = 'Muted: (.*?)\W'
+        muted = re.findall(mutedPattern, streamBlock,
+                           re.DOTALL | re.MULTILINE)[0]
+
+        volumePattern = device + '\): (\d{1,})'
+        volumeLevel = re.findall(
+            volumePattern, streamBlock, re.DOTALL | re.MULTILINE)[0]
+
+        if muted == 'true':
+            muted = True
+        else:
+            muted = False
+
+        volume = round(1/15 * int(volumeLevel), 2)
+
+        return muted, device, volume
+
+    def get_app_id(self, winOutput):
+        """Process sys output and return the current app id."""
+        winOutput = winOutput.splitlines()
+        for line in winOutput:
+            if 'mCurrentFocus' in line:
+                currentApp = line.split(' ')[4].split('/')[0]
+                return currentApp
+
+    def get_app_name(self, app_id):
+        """Return the app name from its id and known apps."""
+        i = 0
+        for app in KNOWN_APPS:
+            if app in app_id:
+                app_name = KNOWN_APPS[app]
+                i += 1
+        if i == 0:
+            app_name = None
+
+        return app_name
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def is_volume_muted(self):
+        """Boolean if volume is currently muted."""
+        return self._muted
+
+    @property
+    def volume_level(self):
+        """Return the volume level."""
+        return self._volume
+
+    @property
+    def source(self):
+        """Return the current playback device."""
+        return self._device
+
+    @property
+    def app_id(self):
```

Comment:
This isn't used.

Reply:
Well the app id and name are correctly showing up as attributes on the device's state, even the app name is integrated and working in the media player interface on the dashboard.

Then the app id would be useful to easily complete the list of known apps for everyone.

Reply:
Sorry, I had forgotten that this property existed.